eZ/Bundle/EzPublishCoreBundle/Imagine/Filter/Loader/OverlayBaseLoader.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace eZ\Bundle\EzPublishCoreBundle\Imagine\Filter\Loader;
+
+use Imagine\Filter\Basic\Fill;
+use Imagine\Image\Fill\Gradient\Horizontal;
+use Imagine\Image\Fill\Gradient\Vertical;
+use Imagine\Image\ImageInterface;
+use Imagine\Image\Point;
+use Imagine\Image\Box;
+use Imagine\Exception\InvalidArgumentException;
+
+class OverlayBaseLoader extends FilterLoaderWrapped
+{
+    public function load(ImageInterface $image, array $options = array())
+    {
+        if (!isset($options['opacity'], $options['startColor'], $options['endColor'], $options['linerClass'])) {
+            throw new InvalidArgumentException('Missing one of required options');
+        }
+        
+        $horizontalLinerClass = Horizontal::class;
+        $verticalLinerClass = Vertical::class;
+        if (!\in_array($options['linerClass'], [$horizontalLinerClass, $verticalLinerClass], true)){
+            throw new InvalidArgumentException(
+                'Unsupported the "linerClass" it should be "' . $horizontalLinerClass . '" or "'
```

Comment:
I think that ` the` should be removed. I'd also change ` it should be` to `, expected: `.